python/ccf/clients.py

```diff
@@ -36,31 +38,37 @@ def truncate(string, max_len=256):
 
 DEFAULT_CONNECTION_TIMEOUT_SEC = 3
 DEFAULT_REQUEST_TIMEOUT_SEC = 3
+DEFAULT_COMMIT_TIMEOUT_SEC = 3
 
 
 class Request:
-    def __init__(
-        self, path, params=None, http_verb="POST", headers=None, params_in_query=None
-    ):
+    def __init__(self, path, params=None, http_verb="POST", headers=None):
         if headers is None:
             headers = {}
 
-        # TODO: remove
-        if params_in_query is None:
-            params_in_query = http_verb == "GET"
-
         self.path = path
         self.params = params
         self.http_verb = http_verb
         self.headers = headers
-        self.params_in_query = params_in_query
+        self.params_in_query = http_verb == "GET" or http_verb == "DELETE"
+
+    def get_params(self):
+        if self.params_in_query and self.params is not None:
+            return f"?{urllib.parse.urlencode(self.params)}"
```

Comment:
We try to parse the values as JSON, so we need `foo="bar"` rather than `foo=bar` (this is why we had `build_query_string`, calling `urlencode` directly won't work). This is a hopefully temporary restriction, but probably not worth tackling in this PR.